ast/compile_test.go

```diff
@@ -1103,14 +1103,7 @@ g(1,2) { true }`,
 p { true }`,
 		"mod6.rego": `package badrules.existserr
 
-p { true }`,
-		"mod7.rego": `package badrules.redeclaration
-
-p1 := 1
-p1 := 2
-
-p2 = 1
-p2 := 2`})
```

Comment:
Should we also add a test case for the happy flow, to avoid regression?

Reply:
Amended the ast commit to include a test case for this.